pkg/controller/core.oam.dev/v1alpha2/application/apply.go

```diff
@@ -93,41 +94,9 @@ func (h *AppHandler) initDispatcher() {
 	}
 }
 
-// ApplyAppManifests will dispatch Application manifests
-func (h *AppHandler) ApplyAppManifests(ctx context.Context, comps []*types.ComponentManifest, policies []*unstructured.Unstructured) error {
-	// dispatch workload in policy before workflow start
-	if len(policies) != 0 {
-		if err := h.Dispatch(ctx, policies...); err != nil {
-			return errors.WithMessage(err, "cannot dispatch policies")
-		}
-	}
-
-	appRev := h.currentAppRev
-	if (h.app.Spec.Workflow != nil && len(h.app.Spec.Workflow.Steps) > 0) || h.app.Annotations[oam.AnnotationAppRevisionOnly] == "true" {
-		return h.createResourcesConfigMap(ctx, appRev, comps, policies)
-	}
-	if appWillRollout(h.app) {
-		return nil
-	}
-
-	// dispatch packaged workload resources before dispatching assembled manifests
-	for _, comp := range comps {
-		if len(comp.PackagedWorkloadResources) != 0 {
-			if err := h.Dispatch(ctx, comp.PackagedWorkloadResources...); err != nil {
-				return errors.WithMessage(err, "cannot dispatch packaged workload resources")
-			}
-		}
-		if comp.InsertConfigNotReady {
-			continue
-		}
-	}
-	a := assemble.NewAppManifests(h.currentAppRev).WithWorkloadOption(assemble.DiscoveryHelmBasedWorkload(ctx, h.r.Client))
-	manifests, err := a.AssembledManifests()
-	if err != nil {
-		return errors.WithMessage(err, "cannot assemble application manifests")
-	}
-	_, err = h.DispatchAndGC(ctx, manifests...)
-	return err
+// ProduceArtifacts will produce Application artifacts that will be saved in configMap.
+func (h *AppHandler) ProduceArtifacts(ctx context.Context, comps []*types.ComponentManifest, policies []*unstructured.Unstructured) error {
+	return h.createResourcesConfigMap(ctx, h.currentAppRev, comps, policies)
```

Comment:
let's remove configmap, use in memory structure.

Reply:
if it's not used, let's delete the code

Reply:
ProduceArtifacts is not real dead code.This function was used extensively in the  revision_test.go.